tests/test_core.py

```diff
@@ -3,11 +3,97 @@
 import numpy as np
 
 from pymodulon.core import IcaData
+from pymodulon.io import load_json_model, save_to_json
 
 
 LOGGER = logging.getLogger(__name__)
 
 
+def test_init_thresholds(ecoli_obj, mini_obj_opt, caplog, tmp_path):
```

Comment:
We might be able to utilize pytest 'parameterize' decorator here instead of the nested for-loop. If you haven't used it before I can take a crack at it sometime in the future.